creme/test_.py

```diff
@@ -130,3 +136,58 @@ def is_estimator(obj):
 )
 def test_check_estimator(estimator):
     utils.estimator_checks.check_estimator(estimator)
+
+
+@pytest.mark.parametrize(
+    'optimizer, initializer',
+    itertools.product([
+        pytest.param(copy.deepcopy(optimizer), id=str(optimizer))
+            for optimizer in [
+                optim.AdaBound(),
+                optim.AdaDelta(),
+                optim.AdaGrad(),
+                optim.AdaMax(),
+                optim.Adam(),
+                optim.Momentum(),
+                optim.NesterovMomentum(),
+                optim.RMSProp(),
+                optim.SGD()
+            ]
+        ],
+        [
+            pytest.param(copy.deepcopy(initializer), id=str(initializer))
+            for initializer in [
+                optim.initializers.Zeros(),
+                optim.initializers.Constant(value=3.14),
+                optim.initializers.Normal(mu=0, sigma=1, random_state=42)
+            ]
+        ]
+    )
+)
+def test_check_gradient(optimizer, initializer):
+    # Test gradient finite difference with linear regression
+    X_y = stream.iter_sklearn_dataset(
+        dataset=datasets.load_boston(),
+        shuffle=True,
+        random_state=42
+    )
+
+    scaler = preprocessing.StandardScaler()
+    model = linear_model.LinearRegression()
+    model.weights = initializer(shape = len(model.weights))
+
+    for x, y in X_y:
+        x = scaler.fit_one(x).transform_one(x)
+        utils.estimator_checks.check_gradient_finite_difference(model, x, y, delta = 0.001)
+        model.fit_one(x, y)
+        
+    # Test gradient finite difference with logistic regression
+    X_y = creme.datasets.fetch_electricity()
+    scaler = preprocessing.StandardScaler()
+    model = linear_model.LogisticRegression()
+    model.weights = initializer(shape = len(model.weights))
```

Comment:
This is not what I meant. I was thinking of replacing `self.weights = collections.defaultdict(float)` by `self.weights = collections.defaultdict(initializer)` in the `GLM` class. `initializer` would thus be a parameter specified in the `GLM` signature.

Reply:
@MaxHalford,

I updated GLM class, LinearRegression and LogisticRegression classes by adding a new parameter : initializer. This parameter is defined as optim.initializers.Zeros() by default in LinearRegression and LogisticRegression classes too. 

I updated the initializers by setting the default shape to 1 when it is called. It's easier for linear models and it makes sense to me. (Easier because of the defaultdict scheme with weights).

Did I understand what you wanted for the GLM class, the intializers and the gradient test? 

Thank you for your review.

Reply:
Yes it looks good! Are the unit tests passing?